packages/store/addon/-private/system/model/internal-model.ts

```diff
@@ -1240,20 +1258,48 @@ export default class InternalModel {
     @method adapterDidInvalidate
     @private
   */
-  adapterDidInvalidate(errors) {
-    let attribute;
+  adapterDidInvalidate(parsedErrors, error) {
+    if (false) {
+      let attribute;
+
+      for (attribute in parsedErrors) {
+        if (parsedErrors.hasOwnProperty(attribute)) {
+          this.addErrorMessageToAttribute(attribute, parsedErrors[attribute]);
+        }
+      }
+
+      this.send('becameInvalid');
 
-    for (attribute in errors) {
-      if (errors.hasOwnProperty(attribute)) {
-        this.addErrorMessageToAttribute(attribute, errors[attribute]);
+      this._recordData.commitWasRejected();
+    } else {
+      let attribute;
+      if (error && parsedErrors) {
+        let jsonApiErrors: JsonApiValidationError[] = errorsHashToArray(parsedErrors);
+        this.send('becameInvalid');
+        if (jsonApiErrors.length === 0) {
+          jsonApiErrors = [{ title: 'Invalid Error', detail: '', source: { pointer: '/data' } }];
+        }
+        this._recordData.commitWasRejected({}, jsonApiErrors);
+      } else {
+        this.send('becameError');
+        this._recordData.commitWasRejected({});
       }
     }
+  }
 
-    this.send('becameInvalid');
+  notifyErrorsChange() {
+    let jsonApiErrors = this._recordData.getErrors() || [];
+    // TODO NOW tighten this check
+    // don't think we need this
+    let invalidErrors: JsonApiValidationError[] = jsonApiErrors.filter((err) => err.source !== undefined && err.title !== undefined) as JsonApiValidationError[];
```

Comment:
why are we forced to cast here? why would we need this filter? ( is this where userland errors enter the system?)

Reply:
Left a note fore myself above ^^ looking into this right now

Reply:
This was leftover code before adapter errors were handled by request state service. removed